### PR TITLE
Circuit trait impl for BlindBid circuit

### LINKED ISSUE #65: Implement CircuitBuilder trait for blindbidproof
In order to be able to work more comfortably from rusk perspective in https://github.com/dusk-network/rusk/pull/80 it would be nice to implement the latest feature added to the plonk repo to see how it fits and if it's a good solution (see: https://github.com/dusk-network/plonk/pull/303).

If it works well and we can export this `CircuitBuilder` impl for the blindbidproof circuit, that would solve a lot of our problems.

### PR DESCRIPTION
**Pull request by CPerezz**
Closes #65

### CODE REVIEW

File: src/proof.rs [hunk 1]
```diff
@@ -22,166 +22,357 @@ use poseidon252::{
...
+    pub size: usize,
+    pub pi_constructor: Option<Vec<PublicInput>>,
+}
+
+impl<'a> Circuit<'a> for BlindBidCircuit<'a> {
+    fn gadget(
+        &mut self,
+        composer: &mut StandardComposer,
+    ) -> Result<Vec<PublicInput>, Error> {
+        // Instanciate PI vector.
```
Comment (Bounce23) [inline src/proof.rs:47]: Nit. *instantiate

File: src/proof.rs [hunk 2]
```diff
@@ -22,166 +22,357 @@ use poseidon252::{
...
+            Point::from_private_affine(
+                composer,
+                bid.stealth_address.pk_r().into(),
+            ),
+            Point::from_private_affine(
+                composer,
+                bid.stealth_address.R().into(),
+            ),
+        );
+        let bid_elegibility_ts =
```
Comment (Bounce23) [inline src/proof.rs:96]: Nit. *eligibility.

File: src/proof.rs [hunk 3]
```diff
@@ -22,166 +22,357 @@ use poseidon252::{
...
+            composer,
+            branch.clone(),
+            bid_hash.var,
+            branch.root,
+        );
+        // Add PI constraint for the root to the PI constructor
+        pi.push(PublicInput::BlsScalar(
+            -branch.root,
+            composer.circuit_size(),
+        ));
```
Comment (Bounce23) [inline src/proof.rs:138]: Is this supposed to be a double closure?

File: src/proof.rs [hunk 4]
```diff
@@ -22,166 +22,357 @@ use poseidon252::{
...
+            bid.c,
+            composer.circuit_size(),
+            composer.circuit_size() + 1,
+        ));
+        // Assert computed_commitment == announced commitment.
+        composer.assert_equal_public_point(computed_c, bid.c);
+
+        // 6. 0 < value <= 2^64 range check
+        // v < 2^64
+        composer.range_gate(bid_value.var, 64usize);
```
Comment (Bounce23) [inline src/proof.rs:217]: Isn't the range function naturally taking in a usize? If so, is the input of it here redundant?

File: src/proof.rs [hunk 5]
```diff
@@ -22,166 +22,357 @@ use poseidon252::{
...
+
+    fn compile(
+        &mut self,
+        pub_params: &PublicParameters,
+    ) -> Result<(ProverKey, VerifierKey, usize), Error> {
+        // Setup PublicParams
+        let (ck, _) = pub_params.trim(1 << 16)?;
+        // Generate & save `ProverKey` with some random values.
+        let mut prover = Prover::new(b"TestCircuit");
+        // Set size & Pi builder
```
Comment (Bounce23) [inline src/proof.rs:294]: Nit. *PI

**Review by Bounce23 — COMMENTED** [review]
Just a few small things. I didn't add a comment each time but its worth changing the misspelling of 'eligibility' or 'eligible', it's often written as variations of 'elegible'.

**Review by vlopes11 — APPROVED** [review]
LGTM, let consider in the future splitting the `fn gadget` into two phases
* Prepare the required data
* The circuit itself

Right now we have these two mixed into a single huge function.

File: src/proof.rs (hunk 3 above)
Comment (CPerezz) [reply src/proof.rs:138]: WDYM by double closure? There's no closure in here. Is just pushing an enum to a vec.

Comment (Bounce23) [reply src/proof.rs:138]: The format on my screen was squashed and messed up, so it placed the braces in the wrong place.

**Pull request merged**